src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -138,14 +138,19 @@ export function TopNavMenu<QT extends AggregateQuery | Query = Query>(
       'kbnTopNavMenu__wrapper--hidden': visible === false,
     });
     if (setMenuMountPoint) {
+      const badgesEl = renderBadges();
+      const menuEl = renderMenu(menuClassName);
       return (
         <>
-          <MountPointPortal setMountPoint={setMenuMountPoint}>
-            <span className={`${wrapperClassName} kbnTopNavMenu__badgeWrapper`}>
-              {renderBadges()}
-              {renderMenu(menuClassName)}
-            </span>
-          </MountPointPortal>
+          {(badgesEl || menuEl) && (
```

Comment:
This stop setting the menu mount point when there is no content for apps that use top nav component